anomalib/models/components/feature_extractors/feature_extractor.py

```diff
@@ -0,0 +1,159 @@
+"""Convenience wrapper for feature extraction methods."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List, Tuple, Union
+
+import torch
+from omegaconf import DictConfig
+from torch import Tensor, nn
+
+from .timm import TimmFeatureExtractor, TimmFeatureExtractorParams
+from .torchfx import TorchFXFeatureExtractor, TorchFXFeatureExtractorParams
+
+
+class FeatureExtractor(nn.Module):
```

Comment:
I have some concerns about the class structure of the feature extractors, mainly caused by the design which involves several layers of wrappers around the feature extractor network.

I feel there are too many objects living in the torch model which are all referred to as 'feature extractor'. First of all, the torch model has a constructor argument named `feature extractor`, which is used to construct the `FeatureExtractor` class, which is then assigned to `self.feature_extractor`. 

```
self.feature_extractor = FeatureExtractor(feature_extractor)
```
This already looks a bit confusing.

Then, due to the wrapper design, this `self.feature_extractor` object now contains several layers of attributes which are all called `feature_extractor`. As a result, the actual network that performs the feature extraction resides at
```
self.feature_extractor.feature_extractor.feature_extractor
```
This makes the code around the feature extraction functionality a bit confusing and hard to navigate.

----
While I understand the motivation behind the design, I wonder if there might be a way to achieve the same thing without relying on this many wrappers. How about using a base `FeatureExtractor` class which has `TorchFXFeatureExtractor` and `TimmFeatureExtractor` as subclasses.